packages/@uppy/tus/src/index.js

```diff
@@ -277,48 +303,82 @@ module.exports = class Tus extends Plugin {
     })
   }
 
+  /**
+   * See the comment on the upload() method.
+   *
+   * Additionally, when an upload is removed, completed, or cancelled, we need to close the WebSocket connection. This is handled by the resetUploaderReferences() function, so the same guidelines apply as in upload().
+   */
   connectToServerSocket (file) {
     return new Promise((resolve, reject) => {
       const token = file.serverToken
       const host = getSocketHost(file.remote.companionUrl)
-      const socket = new Socket({ target: `${host}/api/${token}` })
+      const socket = new Socket({ target: `${host}/api/${token}`, autoOpen: false })
       this.uploaderSockets[file.id] = socket
-      this.uploaderEvents[file.id] = createEventTracker(this.uppy)
+      this.uploaderEvents[file.id] = new EventTracker(this.uppy)
 
       this.onFileRemove(file.id, () => {
+        queuedRequest.abort()
         socket.send('pause', {})
+        this.resetUploaderReferences(file.id)
         resolve(`upload ${file.id} was removed`)
       })
 
       this.onPause(file.id, (isPaused) => {
-        isPaused ? socket.send('pause', {}) : socket.send('resume', {})
+        if (isPaused) {
+          // Remove this file from the queue so another file can start in its place.
+          queuedRequest.abort()
+          socket.send('pause', {})
+        } else {
+          // Resuming an upload should be queued, else you could pause and then resume a queued upload to make it skip the queue.
+          queuedRequest.abort()
+          queuedRequest = this.requests.run(() => {
+            socket.send('resume', {})
+            return () => {}
+          })
+        }
       })
 
-      this.onPauseAll(file.id, () => socket.send('pause', {}))
+      this.onPauseAll(file.id, () => {
+        queuedRequest.abort()
+        socket.send('pause', {})
+      })
 
-      this.onCancelAll(file.id, () => socket.send('pause', {}))
+      this.onCancelAll(file.id, () => {
+        queuedRequest.abort()
+        socket.send('pause', {})
+        this.resetUploaderReferences(file.id)
+        resolve(`upload ${file.id} was canceled`)
+      })
 
       this.onResumeAll(file.id, () => {
+        queuedRequest.abort()
         if (file.error) {
           socket.send('pause', {})
         }
-        socket.send('resume', {})
+        queuedRequest = this.requests.run(() => {
+          socket.send('resume', {})
```

Comment:
We forgot to `return () => {}`, now we get this error:
![image](https://user-images.githubusercontent.com/7578559/63730302-32800600-c884-11e9-8d5a-f745eb03ada3.png)

To reproduce:
1. select some instagram images
2. stop all - resume all - stop all (<--- aand here is the error)

Reply:
:+1: thanks!